pkg/agent/cniserver/pod_configuration_linux.go

```diff
@@ -115,6 +116,12 @@ func (pc *podConfigurator) reconcileMissingPods(ifConfigs []*interfacestore.Inte
 	}
 }
 
+// isStaleInterface returns true if the given namespacedName does not exist in desiredPods, or the OVS interface's ofport
+// is "-1" which means the host interface is disconnected.
+func (pc *podConfigurator) isStaleInterface(desiredPods sets.Set[string], ifaceConfig *interfacestore.InterfaceConfig, namespacedName string) bool {
+	return !desiredPods.Has(namespacedName) || ifaceConfig.OFPort == -1
+}
```

Comment:
IMO, "leaking" `desiredPods` (which is a temporary / internal set used by `reconcile`) to this function is not great.
Maybe it would make more sense to have something like this:
```go
// Linux version:
func (pc *podConfigurator) isInterfaceInvalid(ifaceConfig *interfacestore.InterfaceConfig) bool {
	return ifaceConfig.OFPort == -1
}

// Windows version
func (pc *podConfigurator) isInterfaceInvalid(ifaceConfig *interfacestore.InterfaceConfig) bool {
	return false
}
```
And use it like this in `reconcile`: `if !desiredPods.Has(namespacedName) || isInterfaceInvalid(ifaceConfig)`

Reply:
updated.